src/v/cloud_topics/level_zero/stm/ctp_stm.cc

```diff
@@ -57,6 +61,152 @@ class ctp_stm_consumer {
 ctp_stm::ctp_stm(ss::logger& logger, raft::consensus* raft)
   : raft::persisted_stm<>(name, logger, raft) {}
 
+ss::future<> ctp_stm::start() {
+    ssx::spawn_with_gate(_gate, [this] { return prefix_truncate_below_lro(); });
+    return raft::persisted_stm<>::start();
+}
+
+ss::future<> ctp_stm::stop() {
+    _lro_advanced.broken();
+    _as.request_abort();
+    co_await raft::persisted_stm<>::stop();
+}
+
+ss::future<> ctp_stm::prefix_truncate_below_lro() {
+    static constexpr auto retry_backoff_time = 5s;
+    static constexpr auto min_truncate_period = 60s;
+    while (!_gate.is_closed()) {
+        vlog(
+          _log.trace,
+          "Waiting for LRO to advance past {}, current snapshot index: {}",
+          _state.get_max_collectible_offset(),
+          _raft->last_snapshot_index());
+        try {
+            if (
+              _raft->last_snapshot_index()
+              >= _state.get_max_collectible_offset()) {
+                co_await _lro_advanced.wait();
+            } else {
+                co_await _lro_advanced.wait(retry_backoff_time);
```

Comment:
The body of this while loop feels like something that should maybe be moved into the Raft implementation, given its cuts are so tied to safely truncating a Raft log at a Raft offset. Especially then, seems it can be reused pretty cleanly across here and log_eviction_stm.

Reply:
Yeah fair, I was surprised there was so much logic in eviction stm TBH

Reply:
I will send a followup